lib/sycamore/sycamore/reader.py

```diff
@@ -290,11 +293,18 @@ def opensearch(
         client_params = OpenSearchReaderClientParams(os_client_args=os_client_args)
         query_params = (
             OpenSearchReaderQueryParams(
-                index_name=index_name, query=query, reconstruct_document=reconstruct_document, kwargs=query_kwargs
+                index_name=index_name,
```

Comment:
I would change this to
```
query_parms = OpenSearchReaderQueryParams(index_name=index_name, query=query, reconstruct_document=reconstruct_document, doc_reconstructor=doc_reconstructor, kwargs=query_kwargs)
# and put
if query is None:
    query = DEFAULT_QUERY
# in OpenSearchReaderQueryParams.__init__
```
Otherwise there's lots of duplicated code.

Reply:
Ok